Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RUDecodeData.h

```diff
@@ -98,11 +98,19 @@ int RUDecodeData::decodeROF(const Mapping& mp, const o2::InteractionRecord ir)
       auto chip = mp.getGlobalChipID(cid, cabHW, *this->ruInfo);
       return chip;
     };
+    auto localChipIdGetter = [this, &mp, cabHW](int gid) {
+      auto localID = mp.getLocalChipID(gid, cabHW, *this->ruInfo);
+      return localID;
+    };
+
     int ret = 0;
     // dumpcabledata(icab);
 
-    while ((ret = AlpideCoder::decodeChip(*chipData, cableData[icab], seenChipIDs, chIdGetter)) || chipData->isErrorSet()) { // we register only chips with hits or errors flags set
+    std::vector<uint16_t> seenChipIDsInCable;
+    std::map<int, ChipPixelData*> seenChips;
+    while ((ret = AlpideCoder::decodeChip(*chipData, cableData[icab], seenChipIDsInCable, chIdGetter)) || chipData->isErrorSet()) { // we register only chips with hits or errors flags set
       setROFInfo(chipData, cableLinkPtr[icab]);
+      seenChips[chipData->getChipID()] = chipData;
```

Comment:
Could you move this memory-allocating stuff under `if (verifyDecoder)`?

Reply:
Done